core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -151,6 +151,55 @@ void updateAmounts() throws AnchorException {
     Sep31Transaction txn = Context.get().getTransaction();
     Sep31PostTransactionRequest request = Context.get().getRequest();
     Amount feeResponse = Context.get().getFee();
+    boolean withoutQuote = request.getQuoteId() == null;
+
+    AssetInfo reqAsset = assetService.getAsset(request.getAssetCode(), request.getAssetIssuer());
+    BigDecimal reqAmount = decimal(request.getAmount(), reqAsset);
+    BigDecimal fee = decimal(feeResponse.getAmount(), reqAsset);
+    String feeAsset = String.valueOf(feeResponse.getAsset());
+
+    BigDecimal amountIn;
+    String amountInAsset = reqAsset.getCode();
+    BigDecimal amountOut;
+    String amountOutAsset;
+
+    if (withoutQuote) {
+      boolean strictSend = sep31Config.getPaymentType() == STRICT_SEND;
+      // Without quote.
+      amountOutAsset = amountInAsset;
+      if (strictSend) {
+        // amount_in = req.amount
+        // amount_out = amount_in - amount fee
+        amountIn = reqAmount;
+        amountOut = amountIn.subtract(fee);
+      } else {
+        // amount_in = req.amount + fee
+        // amount_out = req.amount
+        amountIn = reqAmount.add(fee);
+        amountOut = reqAmount;
+      }
+    } else {
+      // With quote
+      Sep38Quote quote = sep38QuoteStore.findByQuoteId(request.getQuoteId());
```

Comment:
You need to handle the case where the quote can't be found. What if a person submits a transaction with a non-existing quote, right?

Reply:
The validations are done before it enters the `updateAmount()` function. 😺 

The case is validated here at [Line#374](https://github.com/stellar/java-stellar-anchor-sdk/blob/2022-05-09-sep31-simplify-amount-fee/core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java#L374).